backend/src/controllers/admin/addCourse.ts

```diff
@@ -0,0 +1,21 @@
+import { NextFunction, Request, Response } from 'express';
+
+import { addCourseProvider } from 'db/providers/courseProvider';
+import { ICreateCourseBody } from 'interfaces/ICourses/IQueryCourses';
+
+const addCourse = async (
+  req: Request<never, never, ICreateCourseBody>,
+  res: Response<never, { results: ICreateCourseBody }>,
+  next: NextFunction,
+) => {
+  try {
+    const newCourse = res.locals.results;
```

Comment:
We should store as the locals.results value that provider returns after creating new document!